code/datums/components/_component.dm

```diff
@@ -405,7 +405,7 @@
 	if(ispath(component_type))
 		component_type = GetExactComponent(component_type)
 	if(!component_type)
-		CRASH("Attempted to remove a null or non-existent component '[component_type]' from '[type]'")
```

Comment:
Why are you removing this?

Reply:
I don't think it is correct to crash under this circumstance because components being removed external to their sources is an expected common occurence which will happen frequently on deletion of the parent object.

Reply:
Alright that seems fair. I've had struggles with it aswell with the shuttle cling deletions